pymongo/asynchronous/monitor.py

```diff
@@ -521,14 +522,21 @@ def _shutdown_monitors() -> None:
     monitor = None
 
 
-def _shutdown_resources() -> None:
+async def _shutdown_resources() -> None:
     # _shutdown_monitors/_shutdown_executors may already be GC'd at shutdown.
     shutdown = _shutdown_monitors
     if shutdown:  # type:ignore[truthy-function]
         shutdown()
     shutdown = _shutdown_executors
     if shutdown:  # type:ignore[truthy-function]
-        shutdown()
+        await shutdown()
+
+
+def _run_shutdown_resources():
+    if _IS_SYNC:
+        _shutdown_resources()
+    else:
+        asyncio.run(_shutdown_resources())
```

Comment:
This won't work because there is already an event loop running, we need something like https://github.com/minrk/asyncio-atexit

Reply:
yeah, something was funky HAHA thanks for showing me a fix!

Reply:
Background: https://discuss.python.org/t/atexit-for-asyncio/13911/10

Reply:
would i be allowed to import this and use the package? or should i just take the code that we need and put in on our own code base? thoughts?

Reply:
Do we need this at all? What happens if we do nothing for async clients at exit?

Reply:
I took it out of async and it seems to be fine so far.